m4/with_unitdir.m4

```diff
@@ -0,0 +1,63 @@
+
+#
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+#
+
+AC_DEFUN([PBS_AC_WITH_UNITDIR],
+[
+  AC_ARG_WITH([unitdir],
+    AS_HELP_STRING([--with-unitdir=DIR],
```

Comment:
Thanks @hirenvadalia for the review.
OS vendors defined the other paths as well i.e.
    /etc/systemd/system for Local configuration
    /run/systemd/system for Runtime units
    /usr/lib/systemd/system for Units of installed packages ..etc
So with the help of "--with-unitdir" user can place the file wherever user wants because i couldn't find any reason to restrict the unit file to one directory only.

Reply:
>  if in future OS vendors change those paths then anyways we have to change this m4 file to adopt new path, right?
You are right. IMHO, Until we update the code to adopt to the changes, a user who is not familiar with the semantics of m4 can make use of this harmless piece of code.

Reply:
@nithinj if the user is not familiar with m4 then they can simply copy the file to correct location. And my view this is a harmful code as it creates confusion.